packages/eslint-plugin/CHANGELOG.md

```diff
@@ -171,6 +171,10 @@
 -   The bundled `eslint-plugin-react` dependency has been updated from requiring `^7.20.0` to requiring `^7.22.0` ([#27965](https://github.com/WordPress/gutenberg/pull/27965)).
 -   The bundled `eslint-plugin-react-hooks` dependency has been updated from requiring `^4.0.4` to requiring `^4.2.0` ([#27965](https://github.com/WordPress/gutenberg/pull/27965)).
 
+### New Features
```

Comment:
This changelog entry is misplaced. It should be moved to `## Unreleased` section. We will have to rethink the way changelogs are maintained, as it's too easy to run into this type of situation when merging the latest changes from `trunk`.

Reply:
> We will have to rethink the way changelogs are maintained,

100%. I was thinking if we could semi-automate the changelog by asking the PR author to simply state:
- which category this falls under (bug fix, enhancements, breaking change...)
- message

The rest can be performed by the automated action at merge time?

Reply:
I was looking at [Jetpack monorepo](https://github.com/Automattic/jetpack) the other day, and they use an interesting system.

They keep a `changelog` folder for every package:

<img width="569" alt="Screenshot 2023-02-13 at 09 33 51" src="https://user-images.githubusercontent.com/699132/218409161-1ad460b5-696c-4095-8384-d16ebf8a2c75.png">

Every time a changelog entry is warranted (I don't know the rules), they create a new file:

<img width="679" alt="Screenshot 2023-02-13 at 09 33 40" src="https://user-images.githubusercontent.com/699132/218409140-902b8a3d-3667-4410-969c-a95a9123408f.png">

Those files are read and deleted at publishing time, which completely removes the risk of merge conflicts of any type.

Reply:
That's interesting!

It could be worth chatting to whoever implemented / maintains that system to understand pros and cons, and if it could work well for our use case. For example, I see that they have a CI action that replies to the PR with a checklist, suggesting that a CHANGELOG entry is a mandatory requirement for all PRs?